### PR TITLE
Implement ko-reranker module

### LINKED ISSUE #190: Add Ko-Reranker
https://huggingface.co/Dongjin-kr/ko-reranker

### PR DESCRIPTION
**Pull request by bwook00**
close #190

### CODE REVIEW

File: autorag/nodes/passagereranker/koreranker.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+
+@passage_reranker_node
+def koreranker(queries: List[str], contents_list: List[List[str]],
+               scores_list: List[List[float]], ids_list: List[List[str]],
+               top_k: int) -> Tuple[List[List[str]], List[List[str]], List[List[float]]]:
+    """
+    Rerank a list of contents based on their relevance to a query using ko-reranker.
+    ko-reranker is a reranker based on korean (https://huggingface.co/Dongjin-kr/ko-reranker).
+    :param queries: The list of queries to use for reranking
```
Comment (vkehfdl1) [inline autorag/nodes/passagereranker/koreranker.py:18]: Press enter between function description and parameter description.
(It is rule of sphinx-apidoc)

File: autorag/nodes/passagereranker/koreranker.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    score_result = list(map(lambda x: x[2], results))
+    return content_result, id_result, score_result
+
+
+async def koreranker_pure(query: str, contents: List[str],
+                          scores: List[float], ids: List[str],
+                          top_k: int, model, tokenizer, device) \
+        -> Tuple[List[str], List[str], List[float]]:
+    """
+    Rerank a list of contents based on their relevance to a query using ko-reranker.
```
Comment (vkehfdl1) [inline autorag/nodes/passagereranker/koreranker.py:48]: Also, press Enter after this line.

File: autorag/nodes/passagereranker/koreranker.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    :param contents: The list of contents to rerank
+    :param scores: The list of scores retrieved from the initial ranking
+    :param ids: The list of ids retrieved from the initial ranking
+    :param top_k: The number of passages to be retrieved
+    :param model: The ko-reranker model to use for reranking
+    :param tokenizer: The tokenizer to use for the model
+    :param device: The device to run the model on (GPU if available, otherwise CPU)
+    :return: tuple of lists containing the reranked contents, ids, and scores
+    """
+    model.to(device)
```
Comment (vkehfdl1) [inline autorag/nodes/passagereranker/koreranker.py:59]: Why don't you convert model to cuda outside pure function?

**Review by vkehfdl1 — CHANGES_REQUESTED** [review]
Finally, did you test this at GPU enabled?
You have to test it because we had problems at 'TART'. you know

File: autorag/nodes/passagereranker/__init__.py [hunk 4]
```diff
@@ -1,3 +1,5 @@
+from .koreranker import koreranker
+from .monot5 import monot5
 from .monot5 import monot5
```
Comment (vkehfdl1) [inline autorag/nodes/passagereranker/__init__.py:3]: duplicated import for monot5?

**Review by Eastsidegunn — DISMISSED** [review]
LGTM

**Review by vkehfdl1 — CHANGES_REQUESTED** [review]
If you pass test at GPU, I will approve it.

**Review by Eastsidegunn — APPROVED** [review]

**Review by vkehfdl1 — APPROVED** [review]

**Pull request merged**